### PR TITLE
[CORE-2315] Add created_by to projects

### PR DESCRIPTION
**Pull request by robert-uhl**

### CODE REVIEW

File: src/internal/clusterstate/v2.11.0/project_metadata.go [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
+	"github.com/pachyderm/pachyderm/v2/src/internal/migrations"
+	"github.com/pachyderm/pachyderm/v2/src/internal/pctx"
+)
+
+func addProjectMetadata(ctx context.Context, env migrations.Env) error {
+	ctx = pctx.Child(ctx, "addProjectMetadata")
+	tx := env.Tx
+	if _, err := tx.ExecContext(ctx, `ALTER TABLE core.projects ADD COLUMN created_by TEXT NOT NULL DEFAULT ''`); err != nil {
```
Comment (acohen4) [inline src/internal/clusterstate/v2.11.0/project_metadata.go:14]: I think this is a good opportunity to add a users table under the `auth` schema and link to projects by foreign key

File: src/pfs/pfs.proto [hunk 2]
```diff
@@ -245,6 +245,7 @@ message ProjectInfo {
   AuthInfo auth_info = 3;
   google.protobuf.Timestamp created_at = 4;
   map<string, string> metadata = 5;
+  string created_by = 6;
```
Comment (acohen4) [inline src/pfs/pfs.proto:248]: we should consider as a part of this ticket whether we should make changes in the CLI. Should CreatedBy be presented in any of the commands?

File: src/internal/clusterstate/v2.11.0/project_metadata.go [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+import (
+	"context"
+
+	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
+	"github.com/pachyderm/pachyderm/v2/src/internal/migrations"
+	"github.com/pachyderm/pachyderm/v2/src/internal/pctx"
+)
+
+func addProjectMetadata(ctx context.Context, env migrations.Env) error {
```
Comment (jrockway) [inline src/internal/clusterstate/v2.11.0/project_metadata.go:11]: This is adding createdBy, not metadata.  I'd suggest renaming this function.

**Comment by robert-uhl** [comment]
Closed by #10158

**Pull request closed without merging**